readthedocs/projects/tasks.py

```diff
@@ -1419,29 +1431,36 @@ def clean_build(version_pk):
         return True
 
 
-def _manage_imported_files(version, path, commit, build):
+def _manage_imported_files(version, commit, build):
     """
     Update imported files for version.
 
     :param version: Version instance
-    :param path: Path to search
     :param commit: Commit that updated path
     :param build: Build id
     """
 
+    if not settings.RTD_BUILD_MEDIA_STORAGE:
+        return
+
+    storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
+
     changed_files = set()
+
     # Re-create all objects from the new build of the version
-    for root, __, filenames in os.walk(path):
+    storage_path = version.project.get_storage_path(
+        type_='html', version_slug=version.slug, include_file=False
+    )
+    for root, __, filenames in storage.walk(storage_path):
         for filename in filenames:
             if filename.endswith('.html'):
                 model_class = HTMLFile
             else:
```

Comment:
I'd like to add the `elif project.cdn_enabled` here with an `else: continue`. 👍

Reply:
What does `project.cdn_enabled` do exactly and what would it do differently for search?

Reply:
It means we store _all_ ImportedFile's not just HTMLFile's, so we can purge them from the CDN properly when they change.